pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -108,6 +108,13 @@ func NewFramework(r Registry, plugins *config.Plugins, args []config.PluginConfi
 				if !ok {
 					return nil, fmt.Errorf("plugin %q does not extend prefilter plugin", pf.Name)
 				}
+
+				for _, plugin := range f.preFilterPlugins {
```

Comment:
How about using a map or set to record these plugin names, and use it to check new plugin name?

Reply:
That would be a valid approach, but if it is for optimization, I don't think it is necessary...

Reply:
I suggest we add a validate function like this instead of having a for loop for each set we evaluate in NewFramework:

```
func validatePlugins(plugins *config.Plugins) error {
	if plugins == nil {
		return nil
	}

	validate := func(pgs *config.PluginSet, setName string) error {
		if pgs == nil {
			return
		}
               pgSet := sets.NewString()
		for _, pg := range pgs.Enabled {
                        if pgSet.Has(pg.Name) {
                          return fmt.Errorf("duplicate %q plugin in %q plugin set", pg.Name, setName)
                        }
			pgSet.Insert(pg.Name)
		}
	}
	if err := validate(plugins.QueueSort, "QueueSort"); err != nil {
            return error
        }
	if err:= validate(plugins.PreFilter, "PreFilter"); err != nil {
            return error
        }
        .....
	return nil
}
```

Reply:
I came up with another approach with reflection https://github.com/kubernetes/kubernetes/pull/83243, what do you think of that? If it is a valid approach, I'd like to keep this one as it is. And if not, I could update this PR and close #83243

Reply:
Reflection seems too much. @ahg-g 's suggestion makes more sense to me.

Reply:
I added a comment to the other PR. I think we can use a variation of what you proposed there.